### PR TITLE
feature: add global headers support

### PR DESCRIPTION
**Pull request by joeabrams026**
This PR adds a global headers dictionary to `Settings`, so that there can be headers that apply to all tabs.  
- If a header is defined both globally and on the tab, the tab will have precedence, so that backward compatibility is preserved for existing tabs
- Global headers will flow through to the `COPY CURL` command, again giving precedence to tab headers
- Global headers flow through to the schema introspection request

My first PR here - open to feedback!

set_headers

### CODE REVIEW

File: packages/graphql-playground-react/src/state/sessions/sagas.ts
Comment (joeabrams026) [inline packages/graphql-playground-react/src/state/sessions/sagas.ts]: This will include global headers (and credentials) on the schema fetch that could happen when text changes in the query window.  I'm wondering if there was a reason playground wasn't using `getSessionWithCredentials` here or if it was an oversight.

**Comment by cvburgess** [comment]
Any update on this? It seems like a great addition for testing

**Comment by nenti** [comment]
I would love to help getting this merged. How can I assist moving this forwards?

**Comment by jsnoble** [comment]
When will this get merged? I really needs this functionality now. Having to specify it on the tabs is not a great UX because it loses all history

**Comment by fredvollmer** [comment]
What needs to happen for this to be merged? I believe it would be an extremely valuable feature.

**Comment by robsontenorio** [comment]
Must have :)

**Comment by acao** [comment]
@joeabrams026 hey there, just recently became a maintainer here, want to revive this? or would you rather I handle the conflict resolutions myself?

**Comment by wa4-fearless-otter** [comment]
@acao Please merge this, others and I really need this ;)

File: packages/graphql-playground-react/src/components/Playground.tsx
Comment (joeabrams026) [inline packages/graphql-playground-react/src/components/Playground.tsx:266]: I'm not sure about this merge conflict.  Is it important that headers are `undefined` in this case when there are no `props.headers` ? @acao @yoshiakis

**Comment by joeabrams026** [comment]
I fixed merge conflicts and tested on the preview deploy.  There is one merge line I'm not entirely sure about.  See that comment.  Otherwise I think it can be merged.

**Comment by jbek7** [comment]
@acao any updates on this review?

File: packages/graphql-playground-react/src/components/Playground.tsx
Comment (yoshiakis) [reply packages/graphql-playground-react/src/components/Playground.tsx:266]: Sorry for very late! No, `undefined` here isn't important any more after merging this PR. This `undefined` is for avoiding [this issue](#965) and the changes you've made avoid it too.

File: README.md
Comment (joeabrams026) [reply README.md]: df2b7cc15af02967d8b5e66ea31464f9fdd05133

File: packages/graphql-playground-react/src/state/sessions/sagas.ts
Comment (joeabrams026) [reply packages/graphql-playground-react/src/state/sessions/sagas.ts]: df2b7cc15af02967d8b5e66ea31464f9fdd05133

**Comment by joeabrams026** [comment]
@yoshiakis I believe this is ready for merge

**Comment by yoshiakis** [comment]
@joeabrams026 looks good to me now!

**Review by acao — APPROVED** [review]

**Pull request merged**
